pkg/config/cni.go

```diff
@@ -50,28 +65,60 @@ func MigrateCNIConfig(ctx context.Context, fullState *cluster.FullState, dataDir
 				"iface":   networkConfig.Options[canalFlannelInterface],
 			},
 		}
-		helmChartConfig, err := toHelmChartConfig("rke2-"+canalCNI, canalCfg)
+		helmChartConfig, err = toHelmChartConfig("rke2-"+networkConfig.Plugin, canalCfg)
 		if err != nil {
 			return err
 		}
-		manifestsDir := manifestsDir(dataDir)
-		manifestFile := filepath.Join(manifestsDir, "rke2-canal-config.yaml")
-		err = os.MkdirAll(manifestsDir, 0755)
+	} else if networkConfig.Plugin == calicoCNI {
+		logrus.Infof("Calico CNI plugin is used by RKE1, migrating config to RKE2")
+		calicoCfg := CalicoConfig{
+			Installation: CalicoInstallationSPec{
+				FlexVolumePath: networkConfig.Options[calicoFlexVolumePluginDir],
+				CalicoNetwork: map[string]string{
+					"mtu": strconv.Itoa(networkConfig.MTU),
+				},
+				ControlPlaneNodeSelector: networkConfig.NodeSelector,
+			},
+		}
+		helmChartConfig, err = toHelmChartConfig("rke2-"+networkConfig.Plugin, calicoCfg)
 		if err != nil {
 			return err
 		}
+	} else {
+		logrus.Infof("network plugin is not recognized as rke2 network plugin")
+		return nil
+	}
 
-		// deploy manifest file
-		return ioutil.WriteFile(manifestFile, helmChartConfig, 0600)
+	manifestsDir := manifestsDir(dataDir)
+	manifestFile := filepath.Join(manifestsDir, "rke2-"+networkConfig.Plugin+"-config.yaml")
+	err = os.MkdirAll(manifestsDir, 0755)
+	if err != nil {
+		return err
 	}
-	return nil
+
+	// deploy manifest file
+	return ioutil.WriteFile(manifestFile, helmChartConfig, 0600)
 }
 
-func toHelmChartConfig(helmChartName string, values canalConfig) ([]byte, error) {
-	valuesYaml, err := yamlv3.Marshal(&values)
-	if err != nil {
-		return nil, err
+func toHelmChartConfig(helmChartName string, values interface{}) ([]byte, error) {
+	var (
+		valuesYaml []byte
+		err        error
+	)
+	if helmChartName == "rke2-"+canalCNI {
+		valuesConfig := values.(CanalConfig)
```

Comment:
aren't unchecked type assertions a bad thing? This should be fine but it still feels like it would be good to do 
```suggestion
		valuesConfig, ok := values.(CanalConfig)
		if ok {
```